src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private RandomSize size = new RandomSize();
+
+    private Figure circle = new Circle(size.getSize(), colorSupplier.getRandomColor());
+    private Figure isoscelesTrapezoid = new IsoscelesTrapezoid(size.getSize(),
+            size.getSize(), size.getSize(), colorSupplier.getRandomColor());
+    private Figure rectangle = new Rectangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());
+    private Figure square = new Circle(size.getSize(),
+            colorSupplier.getRandomColor());
+    private Figure rightTriangle = new RightTriangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());
+
+    private Figure [] figures = {
+            circle,
+            isoscelesTrapezoid,
+            rectangle,
+            square,
+            rightTriangle
+    };
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(figures.length);
+        return figures[randomIndex];

```

Comment:
The 'getRandomFigure()' method currently returns a pre-instantiated figure from the 'figures' array. Instead, it should create a new figure instance each time it is called, using random parameters for its dimensions and color.